Data/Vector/Generic.hs

```diff
@@ -415,6 +415,16 @@ take :: Vector v a => Int -> v a -> v a
 take n v = unsafeSlice 0 (delay_inline min n' (length v)) v
   where n' = max n 0
 
+-- | /O(1)/ Yield the last @n@ elements without copying. The vector may
+-- contain less than @n@ elements in which case it is returned unchanged.
+takeEnd :: Vector v a => Int -> v a -> v a
+{-# INLINE_FUSED takeEnd #-}
+takeEnd n v = unsafeSlice i (min n' m) v
+  where
+    n' = max n 0
+    m = length v
+    i = max 0 (m - n)
```

Comment:
What happens if `n` is negative? What about doing `max 0 (m - n')` instead? :smile:

Reply:
Done